src/mbgl/tile/geometry_tile.cpp

```diff
@@ -245,11 +245,13 @@ void GeometryTile::setData(std::unique_ptr<const GeometryTileData> data_) {
 void GeometryTile::reset() {
     MLN_TRACE_FUNC();
 
+    if (pending) {
+        observer->onTileAction(id, sourceID, TileOperation::Cancelled);
+        pending = false;
+    }
+
     // Mark the tile as pending again if it was complete before to prevent
     // signaling a complete state despite pending parse operations.
```

Comment:
Is this comment still accurate?

Reply:
Not really, but I don't think it ever truly was in the context of `reset()`. I've revised the comments in this method and cleared the `loaded` state which more accurately expresses what this method does.